### PR TITLE
Svelte [ResultsIndicator]: Do not show forked/archived suggestions

### PR DESCRIPTION
**Pull request by jasonhawkharris**
We initially display the suggestion for the most severe "Skipped" message by default. However, in numerous scenarios, this suggestion pertains to adding "forked:" or "archived:" filters to the search query. Given that these suggestions often lead to outcomes that the user doesn't actually desire, displaying them isn't essential. Users can access these suggestions via the dropdown in the popover if needed.

From now on, if the most severe item in the list of skipped items is one of the following: 
- repository-fork
- excluded-archive

The results indicator will show a general message: `See more details`.

Additionally, I've fixed some spacing issues and optimized/simplified layout wherever I could.

### Light
Screenshot 2024-04-16 at 8 39 34 PM

### Dark
Screenshot 2024-04-16 at 8 40 58 PM

## Test plan
- Ships with passing unit tests. 
- Manually tested all possible states of the component.

### CODE REVIEW

File: client/shared/src/search/stream.ts [hunk 1]
```diff
@@ -252,6 +252,8 @@ export interface Skipped {
         | 'shard-timedout'
         | 'repository-cloning'
         | 'repository-missing'
+        | 'repository-fork'
+        | 'repository-archive'
```
Comment (camdencheek) [inline client/shared/src/search/stream.ts:256]: Did you check that this is something that's actually sent from the backend? In particular, I do not see any reference to `repository-archive` [here](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@d039be88d9748d711ca08b7960093db02a338dd1/-/blob/internal/search/streaming/api/types.go?L56-87)

File: client/web-sveltekit/src/lib/search/resultsIndicator/ResultsIndicator.svelte
Comment (camdencheek) [inline client/web-sveltekit/src/lib/search/resultsIndicator/ResultsIndicator.svelte:1]: Looks like this change might have broken spacing of the timeout message?

CleanShot 2024-04-15 at 16 25 57@2x

**Comment by jasonhawkharris** [comment]
All states:

**Light mode**
Screenshot 2024-04-16 at 4 35 45 PM

**Dark Mode**
Screenshot 2024-04-16 at 4 35 30 PM

**Comment by camdencheek** [comment]
@jasonhawkharris did you push the changes we worked on? I see the new screenshots, but not the new code

File: client/web-sveltekit/src/lib/search/resultsIndicator/SuggestedAction.svelte [hunk 2]
```diff
@@ -9,85 +9,66 @@
...
+
+    function filterAndSortItems(items: Skipped[]): ItemsBySeverity {
+        if (items.length === 0) {
+            return { items, mostSevere: null }
+        }
+
+        const filteredItems = items.filter((item: Skipped) => {
+            const isNotFork = item.reason !== 'repository-fork'
+            const isNotArchive = item.reason !== 'excluded-archive'
+            return isNotFork && isNotArchive
```
Comment (camdencheek) [inline client/web-sveltekit/src/lib/search/resultsIndicator/SuggestedAction.svelte:28]: maybe easier to read?

```suggestion
            return !['repository-fork', 'excluded-archive'].includes(item.reason)
```

**Review by camdencheek — COMMENTED** [review]
Looks like the spacing got messed up a bit. The horizontal spacing around the info icon is no longer equal, and there is a very large empty space to the right.

CleanShot 2024-04-16 at 17 17 19@2x

Also, it looks like the "Taking too long?" message now takes more vertical space, which causes the page to shift when it is enabled/disabled. 

https://github.com/sourcegraph/sourcegraph/assets/12631702/9251bd8a-f6c5-4966-95e5-580b90bb27ff

File: client/web-sveltekit/src/lib/search/resultsIndicator/SuggestedAction.svelte [hunk 3]
```diff
@@ -9,85 +9,66 @@
...
+        padding-right: 0.25rem;
 
         &.error-text {
             background: var(--danger-2);
         }
     }
 
     .more-details {
-        color: var(--gray-06);
+        color: var(--text-muted);
```
Comment (camdencheek) [inline client/web-sveltekit/src/lib/search/resultsIndicator/SuggestedAction.svelte:71]: Nice :)

File: client/web-sveltekit/src/lib/search/resultsIndicator/SuggestedAction.svelte [hunk 4]
```diff
@@ -98,10 +79,17 @@
...
-        color: var(--text-body);
+
+    .code-font {
+        background-color: var(--secondary);
+        border-radius: 3px;
+        font-family: var(--code-font-family);
+        padding-right: 0.25rem;
+        padding-left: 0.25rem;
+        padding-top: 0.25rem;
+        padding-bottom: 0.25rem;
```
Comment (camdencheek) [inline client/web-sveltekit/src/lib/search/resultsIndicator/SuggestedAction.svelte:93]:
```suggestion
        padding: 0.25rem;
```

File: client/web-sveltekit/src/lib/search/resultsIndicator/TimeoutMessage.svelte [hunk 5]
```diff
@@ -2,33 +2,28 @@
...
         display: flex;
         flex-flow: row nowrap;
+        align-items: center;
+        justify-content: flex-end;
+        gap: 0.5rem 0.5rem;
+
+        .info-badge {
+            background-color: var(--warning-2);
+            border-radius: 3px;
+            padding: 0.15rem 0.25rem 0.15rem 0.25rem;
```
Comment (camdencheek) [inline client/web-sveltekit/src/lib/search/resultsIndicator/TimeoutMessage.svelte:26]:
```suggestion
            padding: 0.15rem 0.25rem;
```

**Review by camdencheek — APPROVED** [review]
LGTM pending the spacing fixes.

**Review by camdencheek — APPROVED** [review]

**Pull request merged**